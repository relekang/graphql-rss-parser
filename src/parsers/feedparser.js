const Readable = require('stream').Readable
const FeedParser = require('feedparser')

const { ParserError, NotAFeedError } = require('../errors')

module.exports = function parseString (feed) {
  return new Promise((resolve, reject) => {
    try {
      const feedparser = new FeedParser()
      feedparser.on('error', error => {
        reject(new ParserError(error))
      })

      let parsedFeed
      feedparser.on('readable', function () {
        const meta = this.meta
        if (!parsedFeed) {
          parsedFeed = Object.assign({}, meta, { entries: [], parser: 'FEEDPARSER' })
        }

        let item
        while ((item = this.read())) {
          delete item.meta
          parsedFeed.entries.push(Object.assign({}, item, {pubDate: new Date(item.pubDate).toISOString()}))
        }
      })

      feedparser.on('end', function () {
        resolve(parsedFeed);
      })

      const stream = new Readable()
      stream.pipe(feedparser)
      stream.push(feed)
      stream.push(null)
    } catch (error) {
      console.log({error})
      reject(error)
    }
  })
  .catch(error => {
    if (error.message === 'Not a feed') {
      throw new NotAFeedError(error)
    }
    throw error
  })
}
