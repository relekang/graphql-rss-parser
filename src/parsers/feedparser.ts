import { Readable } from 'stream';
import FeedParser from 'feedparser';
import _debug from 'debug';

import { EmptyParserOutputError, NotAFeedError, ParserError } from '../errors';
import { Item, ParserResponse } from '../types';

const debug = _debug('graphql-rss-parser:parsers:feedparser');

export function parse(feed: string): Promise<ParserResponse> {
  return new Promise<ParserResponse>((resolve, reject) => {
    try {
      debug('starting to parse');
      const feedparser = new FeedParser({});
      feedparser.on('error', (error: Error) => {
        debug('parsing failed with error', error);
        reject(new ParserError(error, 'FEEDPARSER'));
      });

      let meta: FeedParser.Meta;
      const items: Item[] = [];
      feedparser.on('readable', function (this: FeedParser) {
        meta = meta || (this.meta as FeedParser.Meta);

        let item;
        while ((item = this.read())) {
          items.push({
            title: item.title,
            content_html: item.description,
            url: item.link,
            tags: item.categories,
            date_published: item.pubdate
              ? new Date(item.pubdate).toISOString()
              : undefined,
            authors: item.author ? [{ name: item.author }] : [],
            id: item.guid,
          });
        }
      });

      feedparser.on('end', function () {
        debug('done parsing');
        if (!meta) {
          return reject(new EmptyParserOutputError());
        }
        resolve({
          parser: 'FEEDPARSER',
          title: meta.title,
          description: meta.description,
          home_page_url: meta.link,
          feed_url: undefined,
          items,
        });
      });

      const stream = new Readable();
      stream.pipe(feedparser);
      stream.push(feed);
      stream.push(null);
    } catch (error: any) {
      debug('parsing failed with error', error);
      reject(new ParserError(error, 'FEEDPARSER'));
    }
  }).catch((error) => {
    debug('parsing failed with error', error);
    if (
      error.message === 'Not a feed' ||
      error.cause.message === 'Not a feed'
    ) {
      throw new NotAFeedError(error);
    }
    throw error;
  });
}
