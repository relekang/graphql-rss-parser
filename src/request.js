const superagent = require('superagent')

const { EmptyHttpResponseError, NotFoundError } = require('./errors')

module.exports = async function request(url) {
  const response = await superagent(url).buffer()

  if (!response.text) throw new EmptyHttpResponseError()
  if (response.notFound) throw new NotFoundError(url)

  return response.text
}
