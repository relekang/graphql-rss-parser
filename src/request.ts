import axios from 'axios';
import _debug from 'debug';

const debug = _debug('graphql-rss-parser:request');

import {
  ConnectionRefusedError,
  DnsLookupError,
  EmptyHttpResponseError,
  UnknownRequestError,
  UpstreamHttpError,
  UpstreamEncryptionError,
  TimeoutError,
} from './errors';

const TIMEOUT = 30 * 1000;

export default async function request(url: string) {
  try {
    debug(`requesting ${url}`);
    const response = await axios({
      url,
      headers: {
        'User-Agent': 'graphql-rss-parser',
      },
      timeout: TIMEOUT,
      responseType: 'arraybuffer',
      transformResponse: undefined,
    });
    debug(`response from ${url} status-code=${response.status}`);
    if (!/2\d\d/.test(response.status.toString())) {
      throw new UpstreamHttpError('Not found', response.status);
    }
    if (!response.data) {
      throw new EmptyHttpResponseError();
    }
    return {
      text: response.data.toString(),
      status: response.status,
      contentType: response.headers['content-type'],
      headers: response.headers,
    };
  } catch (error: any) {
    debug(`request to ${url} failed with error`, error);
    if (error.response && error.response.status) {
      throw new UpstreamHttpError('Upstream HTTP error', error.response.status);
    }
    if (error.code === 'ENOTFOUND' || error.code === 'EAI_AGAIN') {
      throw new DnsLookupError();
    }
    if (error.code === 'ECONNREFUSED' || error.code === 'ECONNRESET') {
      throw new ConnectionRefusedError();
    }

    if (error.code === 'ECONNABORTED' || error.code === 'ETIMEDOUT') {
      throw new TimeoutError();
    }

    if (
      error.constructor === EmptyHttpResponseError ||
      error.constructor === UpstreamHttpError
    ) {
      throw error;
    }

    if (/certificate|tls|ssl/.test(error.toString())) {
      throw new UpstreamEncryptionError(error);
    }

    throw new UnknownRequestError(error);
  }
}
