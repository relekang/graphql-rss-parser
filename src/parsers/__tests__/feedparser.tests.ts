/* eslint-env jest */
import { parse } from '../feedparser';
import request from '../../request';

test('should parse string from rolflekang.com/feed.xml', async () => {
  const { text } = await request('https://rolflekang.com/feed.xml');

  expect(await parse(text)).toMatchSnapshot();
});

test('should parse string from google.blogspot.com/feeds/posts/default', async () => {
  const { text } = await request(
    'http://google.blogspot.com/feeds/posts/default'
  );

  expect(await parse(text)).toMatchSnapshot();
});
