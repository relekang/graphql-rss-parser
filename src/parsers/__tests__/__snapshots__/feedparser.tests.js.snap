// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse string from rolflekang.com/feed.xml 1`] = `
Object {
  "#ns": Array [
    Object {
      "xmlns:atom": "http://www.w3.org/2005/Atom",
    },
  ],
  "#type": "rss",
  "#version": "2.0",
  "#xml": Object {},
  "@": Array [
    Object {
      "xmlns:atom": "http://www.w3.org/2005/Atom",
    },
  ],
  "atom:link": Object {
    "@": Object {
      "href": "http://rolflekang.com/feed.xml",
      "rel": "self",
      "type": "application/rss+xml",
    },
  },
  "author": null,
  "categories": Array [],
  "cloud": Object {},
  "copyright": null,
  "date": null,
  "description": null,
  "entries": Array [
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2017-01-02T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/lets-run-some-races",
      "image": Object {},
      "link": "http://rolflekang.com/lets-run-some-races",
      "origlink": null,
      "permalink": "http://rolflekang.com/lets-run-some-races",
      "pubDate": "2017-01-02T00:00:00.000Z",
      "pubdate": 2017-01-02T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/lets-run-some-races",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/lets-run-some-races",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Mon, 02 Jan 2017 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Let's run some races this year",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Let's run some races this year",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2017-01-01T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/software-development-on-an-ipad",
      "image": Object {},
      "link": "http://rolflekang.com/software-development-on-an-ipad",
      "origlink": null,
      "permalink": "http://rolflekang.com/software-development-on-an-ipad",
      "pubDate": "2017-01-01T00:00:00.000Z",
      "pubdate": 2017-01-01T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/software-development-on-an-ipad",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/software-development-on-an-ipad",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Sun, 01 Jan 2017 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Software development on an iPad",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Software development on an iPad",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2016-06-10T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/filtering-lint-errors",
      "image": Object {},
      "link": "http://rolflekang.com/filtering-lint-errors",
      "origlink": null,
      "permalink": "http://rolflekang.com/filtering-lint-errors",
      "pubDate": "2016-06-10T00:00:00.000Z",
      "pubdate": 2016-06-10T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/filtering-lint-errors",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/filtering-lint-errors",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Fri, 10 Jun 2016 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Filtering lint errors",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Filtering lint errors",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2015-07-26T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/interrail-summer",
      "image": Object {},
      "link": "http://rolflekang.com/interrail-summer",
      "origlink": null,
      "permalink": "http://rolflekang.com/interrail-summer",
      "pubDate": "2015-07-26T00:00:00.000Z",
      "pubdate": 2015-07-26T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/interrail-summer",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/interrail-summer",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Sun, 26 Jul 2015 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Interrail summer",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Interrail summer",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2015-05-23T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/writing-latex-in-atom",
      "image": Object {},
      "link": "http://rolflekang.com/writing-latex-in-atom",
      "origlink": null,
      "permalink": "http://rolflekang.com/writing-latex-in-atom",
      "pubDate": "2015-05-23T00:00:00.000Z",
      "pubdate": 2015-05-23T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/writing-latex-in-atom",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/writing-latex-in-atom",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Sat, 23 May 2015 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Writing (latex) in Atom",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Writing (latex) in Atom",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2014-12-30T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/rmoq",
      "image": Object {},
      "link": "http://rolflekang.com/rmoq",
      "origlink": null,
      "permalink": "http://rolflekang.com/rmoq",
      "pubDate": "2014-12-30T00:00:00.000Z",
      "pubdate": 2014-12-30T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/rmoq",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/rmoq",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Tue, 30 Dec 2014 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "rmoq - a request mock cache",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "rmoq - a request mock cache",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2014-10-03T15:35:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/django-nopassword-one-point-o",
      "image": Object {},
      "link": "http://rolflekang.com/django-nopassword-one-point-o",
      "origlink": null,
      "permalink": "http://rolflekang.com/django-nopassword-one-point-o",
      "pubDate": "2014-10-03T15:35:00.000Z",
      "pubdate": 2014-10-03T15:35:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/django-nopassword-one-point-o",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/django-nopassword-one-point-o",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Fri, 03 Oct 2014 15:35:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "django-nopassword reached 1.0",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "django-nopassword reached 1.0",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2014-05-07T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/moving-to-pelican",
      "image": Object {},
      "link": "http://rolflekang.com/moving-to-pelican",
      "origlink": null,
      "permalink": "http://rolflekang.com/moving-to-pelican",
      "pubDate": "2014-05-07T00:00:00.000Z",
      "pubdate": 2014-05-07T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/moving-to-pelican",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/moving-to-pelican",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Wed, 07 May 2014 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Moving the blog to pelican",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Moving the blog to pelican",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2014-02-02T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/building-tumblr-theme-with-grunt",
      "image": Object {},
      "link": "http://rolflekang.com/building-tumblr-theme-with-grunt",
      "origlink": null,
      "permalink": "http://rolflekang.com/building-tumblr-theme-with-grunt",
      "pubDate": "2014-02-02T00:00:00.000Z",
      "pubdate": 2014-02-02T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/building-tumblr-theme-with-grunt",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/building-tumblr-theme-with-grunt",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Sun, 02 Feb 2014 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Building a Tumblr theme with GruntJS",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Building a Tumblr theme with GruntJS",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2013-10-24T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/backup-routines-for-redisdb",
      "image": Object {},
      "link": "http://rolflekang.com/backup-routines-for-redisdb",
      "origlink": null,
      "permalink": "http://rolflekang.com/backup-routines-for-redisdb",
      "pubDate": "2013-10-24T00:00:00.000Z",
      "pubdate": 2013-10-24T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/backup-routines-for-redisdb",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/backup-routines-for-redisdb",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Thu, 24 Oct 2013 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Backup routine for Redis",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Backup routine for Redis",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2013-05-22T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/django-app-custom-user",
      "image": Object {},
      "link": "http://rolflekang.com/django-app-custom-user",
      "origlink": null,
      "permalink": "http://rolflekang.com/django-app-custom-user",
      "pubDate": "2013-05-22T00:00:00.000Z",
      "pubdate": 2013-05-22T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/django-app-custom-user",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/django-app-custom-user",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Wed, 22 May 2013 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Making your Django app ready for custom users",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Making your Django app ready for custom users",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2013-03-03T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/stats-are-fun",
      "image": Object {},
      "link": "http://rolflekang.com/stats-are-fun",
      "origlink": null,
      "permalink": "http://rolflekang.com/stats-are-fun",
      "pubDate": "2013-03-03T00:00:00.000Z",
      "pubdate": 2013-03-03T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/stats-are-fun",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/stats-are-fun",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Sun, 03 Mar 2013 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Stats are fun",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Stats are fun",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2013-02-26T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/alfred-scripts-for-jekyll",
      "image": Object {},
      "link": "http://rolflekang.com/alfred-scripts-for-jekyll",
      "origlink": null,
      "permalink": "http://rolflekang.com/alfred-scripts-for-jekyll",
      "pubDate": "2013-02-26T00:00:00.000Z",
      "pubdate": 2013-02-26T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/alfred-scripts-for-jekyll",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/alfred-scripts-for-jekyll",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Tue, 26 Feb 2013 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Alfred scripts for Jekyll",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Alfred scripts for Jekyll",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2013-01-27T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/use-array-in-postgresql-with-django",
      "image": Object {},
      "link": "http://rolflekang.com/use-array-in-postgresql-with-django",
      "origlink": null,
      "permalink": "http://rolflekang.com/use-array-in-postgresql-with-django",
      "pubDate": "2013-01-27T00:00:00.000Z",
      "pubdate": 2013-01-27T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/use-array-in-postgresql-with-django",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/use-array-in-postgresql-with-django",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Sun, 27 Jan 2013 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Use array in postgresql with django",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Use array in postgresql with django",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2013-01-03T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/apply-puppet-configuration-automatically",
      "image": Object {},
      "link": "http://rolflekang.com/apply-puppet-configuration-automatically",
      "origlink": null,
      "permalink": "http://rolflekang.com/apply-puppet-configuration-automatically",
      "pubDate": "2013-01-03T00:00:00.000Z",
      "pubdate": 2013-01-03T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/apply-puppet-configuration-automatically",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/apply-puppet-configuration-automatically",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Thu, 03 Jan 2013 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Apply puppet automatically",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Apply puppet automatically",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2012-12-02T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/reading-wikipedia",
      "image": Object {},
      "link": "http://rolflekang.com/reading-wikipedia",
      "origlink": null,
      "permalink": "http://rolflekang.com/reading-wikipedia",
      "pubDate": "2012-12-02T00:00:00.000Z",
      "pubdate": 2012-12-02T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/reading-wikipedia",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/reading-wikipedia",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Sun, 02 Dec 2012 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "Readable Wikipedia",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "Readable Wikipedia",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2012-09-28T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/django-nopassword",
      "image": Object {},
      "link": "http://rolflekang.com/django-nopassword",
      "origlink": null,
      "permalink": "http://rolflekang.com/django-nopassword",
      "pubDate": "2012-09-28T00:00:00.000Z",
      "pubdate": 2012-09-28T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/django-nopassword",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/django-nopassword",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Fri, 28 Sep 2012 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "django-nopassword",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "django-nopassword",
    },
    Object {
      "author": null,
      "categories": Array [],
      "comments": null,
      "date": 2012-09-16T00:00:00.000Z,
      "description": null,
      "enclosures": Array [],
      "guid": "http://rolflekang.com/the-github-lamp",
      "image": Object {},
      "link": "http://rolflekang.com/the-github-lamp",
      "origlink": null,
      "permalink": "http://rolflekang.com/the-github-lamp",
      "pubDate": "2012-09-16T00:00:00.000Z",
      "pubdate": 2012-09-16T00:00:00.000Z,
      "rss:@": Object {},
      "rss:description": Object {
        "@": Object {},
      },
      "rss:guid": Object {
        "#": "http://rolflekang.com/the-github-lamp",
        "@": Object {
          "ispermalink": "true",
        },
      },
      "rss:link": Object {
        "#": "http://rolflekang.com/the-github-lamp",
        "@": Object {},
      },
      "rss:pubdate": Object {
        "#": "Sun, 16 Sep 2012 00:00:00 GMT",
        "@": Object {},
      },
      "rss:title": Object {
        "#": "The Github lamp",
        "@": Object {},
      },
      "source": Object {},
      "summary": null,
      "title": "The Github lamp",
    },
  ],
  "favicon": null,
  "generator": "rl.com v1.0.0",
  "image": Object {},
  "language": null,
  "link": "http://rolflekang.com/",
  "parser": "FEEDPARSER",
  "pubDate": null,
  "pubdate": null,
  "rss:@": Object {},
  "rss:description": Object {
    "@": Object {},
  },
  "rss:generator": Object {
    "#": "rl.com v1.0.0",
    "@": Object {},
  },
  "rss:link": Object {
    "#": "http://rolflekang.com/",
    "@": Object {},
  },
  "rss:title": Object {
    "#": "Writings of Rolf Erik Lekang",
    "@": Object {},
  },
  "title": "Writings of Rolf Erik Lekang",
  "xmlUrl": "http://rolflekang.com/feed.xml",
  "xmlurl": "http://rolflekang.com/feed.xml",
}
`;
